pg_fossa--1.6--1.7.sql

```diff
@@ -0,0 +1,98 @@
+CREATE OR REPLACE FUNCTION fossa_version() RETURNS VARCHAR as $$
+BEGIN
+  RETURN '1.7';
+END; $$ LANGUAGE PLPGSQL IMMUTABLE;
+
+CREATE OR REPLACE FUNCTION fossa_edges(locator VARCHAR(255), filter_tags VARCHAR[], filter_excludes VARCHAR[], filter_origin_paths VARCHAR[], filter_all_origin_paths VARCHAR[], filter_unresolved BOOLEAN, maxdepth INT) RETURNS fossa_edge[] AS $$
+DECLARE
+  current_depth INT := 2;
+  results fossa_edge[] := ARRAY(SELECT CAST(ROW(
+      d.parent,
+      fossa_child(d.child, d.unresolved_locator),
+      fossa_resolved(d.child),
+      d.origin_paths,
+      d.transitive_excludes,
+      d.tags,
+      d.manual,
+      d.is_submodule,
+      1
+    ) AS fossa_edge)
+    FROM "Dependencies" AS d
+    WHERE d.parent=locator
+      AND d.parent != d.child
+      AND (filter_excludes IS NULL OR d.child != ALL(filter_excludes))
+      AND (NOT filter_unresolved OR fossa_resolved(d.child))
+      AND (d.manual OR filter_tags IS NULL OR d.child NOT LIKE 'mvn+%' OR d.unresolved_locator NOT LIKE 'mvn+%' OR d.child LIKE 'mvn+%' AND d.tags && filter_tags OR d.unresolved_locator LIKE 'mvn+%' AND d.tags && filter_tags));
+  intermediate_edges fossa_edge[] := results;
+  working_edges fossa_edge[] := results;
+BEGIN
+  WHILE current_depth <= maxdepth LOOP
+    EXIT WHEN ARRAY_LENGTH(working_edges, 1) = 0;
+    EXIT WHEN ARRAY_LENGTH(working_edges, 1) IS NULL;
+
+    intermediate_edges := ARRAY(
+      SELECT CAST(ROW(
+        d.parent,
+        fossa_child(d.child, d.unresolved_locator),
+        fossa_resolved(d.child),
+        array_union(array_union_agg(d.origin_paths), array_union_agg((w).origin_paths)),
+        array_intersect(array_intersect_agg(d.transitive_excludes), array_intersect_agg((w).excludes)),
+        array_union(array_union_agg(d.tags), array_union_agg((w).tags)),
+        bool_or(d.manual),
+        bool_or(d.is_submodule),
+        min(current_depth)
+      ) AS fossa_edge)
+      FROM "Dependencies" AS d, UNNEST(working_edges) AS w
+      WHERE d.parent = (w).child
+        AND d.parent != d.child
```

Comment:
Added this line

Reply:
Is there a way we can prevent loops where the loop is larger than (parent) <-> (child)? It may be more effort than it's worth with max depth included)

Reply:
We could check that the path we're generating isn't a subset of any existing path. I think with reasonable max depth it might not be worth the effort.